### PR TITLE
tests: speedup lock tests

### PR DESCRIPTION
**Pull request by skshetry**
Patched to reduce timeout from default 5s to 0.01s

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

#### Before
```console
$ pytest tests/func/test_lock.py
2 passed in 10.20s
```

#### After
```console
$ pytest tests/func/test_lock.py
2 passed in 0.20s
```

### CODE REVIEW

**Comment by skshetry** [comment]
@efiop, why is there a 5s sleep? I see that it is from a very old commit 068f7428522f95c2. Is there any reason why we don't retry it in a 0.1s or so?

https://github.com/iterative/dvc/blob/bc7d952e88553eda33bb377a333ab018a1e854ef/dvc/lock.py#L58-L63

File: tests/func/test_lock.py [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-        lockfile = os.path.join(self.dvc.dvc_dir, "tmp", "lock")
-        lock = Lock(lockfile)
-        with lock:
-            with self.assertRaises(LockError):
-                lock2 = Lock(lockfile)
-                with lock2:
-                    pass
+def test_with(tmp_dir, dvc, mocker):
+    # patching to speedup tests
+    mocker.patch("dvc.lock.DEFAULT_TIMEOUT", 0.01)
```
Comment (skshetry) [inline tests/func/test_lock.py:9]: Reduced it further to 0.01.

**Review by pared — COMMENTED** [review]
LGTM, though I have to admit that I don't recall why we had 5s timeout.

**Comment by efiop** [comment]
@skshetry Indeed, we could reduce the timeout, but increase the number of tries. 5s was just a random number that we thought was ok to sleep for.

File: tests/func/test_lock.py (hunk 1 above)
Comment (efiop) [reply tests/func/test_lock.py:9]: So maybe let's just adjust it in the code as discussed above?

Comment (skshetry) [reply tests/func/test_lock.py:9]: This patch will be required after that too unless we decrease the timeout to a very low value.

Let's say we reduced the timeout to 2s. As these tests are negative tests, they will wait for 2s anyway.

Comment (efiop) [reply tests/func/test_lock.py:9]: @skshetry Good point, so we will have to patch both timeout and the number of tries.

**Review by efiop — APPROVED** [review]

**Pull request merged**